include/swift/Basic/LangOptions.h

```diff
@@ -493,6 +493,10 @@ namespace swift {
     /// function bodies.
     bool SkipNonInlinableFunctionBodies = false;
 
+    /// Indicate that the type checker should skip type-checking all
+    /// function bodies, where not otherwise required for type inference.
+    bool SkipAllFunctionBodies = false;
```

Comment:
Would it make sense to scoop these two bools into an enum?

```
enum class FunctionBodySkipping : uint8_t {
  None,
  NonInlinable,
  All
}
```

Reply:
As far as I understand it, the idea is that `SkipNonInlinableFunctionBodies` will actually be the default at some point, with no CLI option. So making it an enum here would make that harder to do in the future. However if that's not true, I'm happy to change it to an enum instead.

Reply:
I think we only ever want to skip non-inlinable function bodies when building a swiftinterface file or TBD. Consolidating these flags into an enum makes sense to me.

Reply:
Sorry, good point. I was just thinking about the flag itself. I'll add the enum, thanks :)

Reply:
I've put the enum under swift in LangOptions.h for now and I'm currently importing it in SILOptions.h. Does that seem okay @harlanhaskins? Alternatively, would you prefer either:
1. a separate header with just that enum, or
2. another enum defined in SILOptions.h?

Reply:
Another enum in SILOptions.h is fine, thank you!

Reply:
Spoke with Harlan privately - ended up deciding on a separate header. Thanks :)!